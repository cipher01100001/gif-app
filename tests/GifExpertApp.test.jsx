import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';


describe('Pruebas en <GifExpertApp />', () => {

    const newCategory = 'Gravity Falls';

    test('should add new categories', () => {
        
        const { container } = render(<GifExpertApp />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: newCategory } });
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: newCategory + '2' } });
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: newCategory + '3' } });
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(3);
        expect(container).toMatchSnapshot();
    });

    test('Should not add a repeated category', () => {

        const { container } = render(<GifExpertApp />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: newCategory } });
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: newCategory } });
        fireEvent.submit(form);

        expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(1);
        expect(container).toMatchSnapshot();
    });

});