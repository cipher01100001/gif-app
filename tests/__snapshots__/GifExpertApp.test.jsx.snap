// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Should not add a repeated category 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Gravity Falls
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en <GifExpertApp /> should add new categories 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Gravity Falls3
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Gravity Falls2
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Gravity Falls
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
